core/services/beacon/settings.py

```diff
@@ -0,0 +1,110 @@
+import json
+import re
+import socket
+from typing import Any, Dict
+
+import psutil
+from commonwealth.settings import settings
+from pykson import (
+    IntegerField,
+    JsonObject,
+    ListField,
+    MultipleChoiceStringField,
+    ObjectListField,
+    StringField,
+)
+
+
+class DefaultSettings(JsonObject):
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+
+class Interface(JsonObject):
+    name = StringField()
+    domain_names = ListField(item_type=str)
+    advertise = ListField(item_type=str)
+    ip = StringField()
+
+    def get_phys(self) -> psutil._common.snicaddr:
+        """
+        return Physical interface from psutil
+        """
+        return psutil.net_if_addrs()[self.name]
```

Comment:
What happens if the interface does not exist anymore ? Like if we remove an ethernet adapter.

Reply:
this code runs from checking the available interfaces, so it can't fail right here. That does open the question of how python-zeroconf handles it, though, which I haven't tested yet

Reply:
that said, I don't think we need to support hot-plugging of ethernet adapters.